packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -188,7 +188,14 @@ function DocSidebarItemLink({
           onClick: onItemClick,
         })}
         {...props}>
-        {label}
+        {isInternalUrl(href) ? (
+          label
+        ) : (
+          <span>
```

Comment:
This `span` element is required for the proper positioning of the icon (in the context of a text link).